subscriber/subscriber.go

```
@@ -198,6 +199,11 @@ func (s *Subscriber) Sync(ctx context.Context) error {
 		}
 	}
 
+	// post-sync actions
+	if err := s.contracts.ProcessActions(ctx); err != nil {
```

If the context gets cancelled we'll break out of the sync loop and this'll post process the actions and immediately return with `context.Cancelled` which we'll error log... we should probably `return` in the for loop or ignore the error. Before this change we also debug log we're synced right before shutting down apparenlty.